src/actions.ts

```diff
@@ -40,71 +47,81 @@ export function filters(options: Options): (item: Item) => boolean {
   return (item: Item) => true;
 }
 
+function makeBulk(
+  bulkOperation: (items: Map<Item, number>) => boolean,
+  baseAmount?: (item: Item) => number
+) {
+  const items = new Map<Item, number>();
+
+  return (options: Options) => ({
+    action: (item: Item) =>
+      items.set(
+        item,
+        baseAmount && options.stock
+          ? Math.min(Math.max(0, (options.stock ?? 0) - shopAmount(item)), amount(item, options))
+          : amount(item, options)
+      ),
+    finalize: () => {
+      bulkOperation(items);
+    },
+  });
+}
+
+function makeMall(mallOptions?: { price?: (item: Item) => number; stock?: boolean }) {
+  return (options: Options) => {
+    const mallItems = new Map<Item, { quantity?: number; limit?: number; price: number }>();
+    const quantity =
+      options.stock && mallOptions?.stock
+        ? (item: Item) =>
+            Math.min(Math.max(0, (options.stock ?? 0) - shopAmount(item)), amount(item, options))
+        : (item: Item) => amount(item, options);
+
+    return {
+      action: (item: Item) =>
+        mallItems.set(item, {
+          quantity: quantity(item),
+          limit: options.limit,
+          price: mallOptions?.price ? mallOptions.price(item) : options.price ?? 0,
```

Comment:
```suggestion
          price: mallOptions?.price?.(item) ?? options.price ?? 0,
```
or whatever you want